### PR TITLE
Basic reliability fix

### LINKED ISSUE #15506: R&D objects are stuck at default reliability
It seems like doing research isn't raising reliability like it should. For example, pico manipulators should normally be at least at 78 reliability after research is done, but now they're still stuck at their default value of 73 reliability. Researching the previous tier of parts also isn't raising their reliability anymore either.
broke

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #15506

I'll probably be removing reliability soon. But this is fix for now.
The level check if was removed to preserve current functionality since it compared number to datum which always passed.

### CODE REVIEW

**Comment by Miauw62** [comment]
>i'll probably be removing reliability soon

based anturk

File: code/modules/research/research.dm [hunk 1]
```diff
@@ -129,14 +129,14 @@ research holder datum.
 
 /datum/research/proc/UpdateDesigns(obj/item/I, list/temp_tech)
 	for(var/T in temp_tech)
```
Comment (neersighted) [inline code/modules/research/research.dm]: Whoa can you unnest this at all?

Comment (AnturK) [inline code/modules/research/research.dm]: Sure

**Comment by Ultimate-Chimera** [comment]
What was reliability even for?

**Comment by AnturK** [comment]
The only noticeable effects are : 
- Advanced Egun failure rate
- Non 100% reliable batteries breaking on emp/normal use
- Not being able to deconstruct higher tier stock parts for research straight away. (relatively)

**Pull request merged**